src/convertToObject.js

```diff
@@ -6,7 +6,27 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const bigArray = sourceString.split(';');
+
+  return bigArray.filter(filt).reduce(callback, {});
```

Comment:
The `reduce` function assumes that every item in `bigArray` is a valid key-value pair. Consider adding error handling or logging for cases where `smallArray` does not have exactly two elements after splitting by ':'.